### DISCUSSION TITLE
Pseudo selector improvement suggestion

### DISCUSSION THREAD

**Discussion by stegano** [tailwindlabs/tailwindcss#8655] [Ideas]
Hi,

Thanks for making such a great library.

When constructing pseudo elements using tailwind, the code seems to be complicated by repeated pseudo selectors.

Therefore, it would be nice if it could be used more simply and clearly by using pseudo selectors and parentheses as shown below.

```tsx
// current
<span
    className={cx(
        "absolute top-1/2 left-7 inline-block w-8 h-5 -translate-y-1/2 text-sm font-medium transition-none transition-all duration-500",
        "after:block after:w-4 after:h-4 after:rounded-2xl after:bg-white after:absolute after:top-0.5 after:-left-5 after:transition-all after:duration-500" // <= here
    )}
>
    {$status}
</span>
```

```tsx
// suggestion
<span
    className={cx(
        "absolute top-1/2 left-7 inline-block w-8 h-5 -translate-y-1/2 text-sm font-medium transition-none transition-all duration-500",
        "after:(block w-4 h-4 rounded-2xl bg-white absolute top-0.5 -left-5 transition-all duration-500)"  // <= here
    )}
    >
    {$status}
</span>
```

Thanks :)

**Comment by 3xau1o** [comment]
Maybe you could checkout [Windi css](https://windicss.org/features/variant-groups.html), has supports for it out of the box.

**Reply by stegano** [reply]
Thanks @mindlid 
That's exactly what I was thinking.
It would be better if this feature was supported in Tailwind as well.

**Reply by 3xau1o** [reply]
Windi CSS generate classes on demand, it means that CSS class names are read then generated, Tailwind on the other side generate the CSS classes then read the class names, and purge not used. The tailwind approach however could be compatible, however it depends on a dev to have enough interest.

**Reply by adamwathan** [reply]
Tailwind generates classes on demand as well. We have experimental support for modifier grouping in the codebase but haven’t decided if we want to actually release it because it leads to bloated CSS files which is bad for your site’s performance.

See this Twitter thread for more details:

https://twitter.com/adamwathan/status/1461519820411789314?s=21&t=7TMUBnQ7F_PZufwmHI3eEQ

Here’s the PR where we added experimental support:

https://twitter.com/adamwathan/status/1528804895444410368?s=21&t=OyZZW17sJHKobSMMuzWOKQ

**Reply by stegano** [reply]
@adamwathan 
Thanks I understand.

**Reply by 3xau1o** [reply]
@adamwathan would it be technically possible and feasible to expand it?

like 

`after:(block w-4 h-4 bg-white)`

to

`after:block after:w-4 after:h-4 after:bg-white`

**Comment by 3xau1o** [comment]
@adamwathan Thanks for the clarification, i understand grouping modifiers bloat the css, if only we could ungroup, and ungrouping would be of the domain of a bundler like webpack, not postcss
